internal/utils/interceptors.go

```diff
@@ -0,0 +1,38 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// MetadataInterceptor is a grpc.UnaryClientInterceptor that extracts the gRPC metadata
+// from the outgoing RPC request context so we can assert on it
+func MetadataInterceptor(t *testing.T, expectedMetadata map[string]string) grpc.UnaryClientInterceptor {
```

Comment:
I spent... too long trying to come up with ways we could mock aspects of the gRPC service to test that things like user-agent and metadata are applied. It was quite challenging.

This should work for validating any metadata we're sending with the outgoing RPC. Validating user-agent was more difficult because `grpc.ClientConn` stores it as a private value until the http2 request is actually constructed.